lib/wal/wal.c

```diff
@@ -209,10 +211,10 @@ wal_timer(struct work_struct *work)
             intvl = max_t(uint64_t, sleep_ns / 10, MSEC_TO_NSEC(1));
             flushc = wal_bufset_flushoff(wal->wbs, WAL_BUF_MAX, flushv);
 
-            while (cv_timedwait(&wal->timer_cv, &wal->timer_mutex,
-                                NSEC_TO_MSEC(intvl), "waltmslp") == ETIMEDOUT &&
-                   !wal->sync_pending && (get_time_ns() - tstart < sleep_ns) &&
-                   !atomic_read(&wal->closing)) {
+            while (!atomic_read(&wal->closing) &&
+                   (cv_timedwait(&wal->timer_cv, &wal->timer_mutex,
+                                NSEC_TO_MSEC(intvl), "waltmslp") == ETIMEDOUT) &&
+                   !wal->sync_pending && (get_time_ns() - tstart < sleep_ns)) {
```

Comment:
This is difficult to understand.   Maybe break it up, dunno, something like:

```
while (!closing) { 
    rc = cv_timedwait();
    if (rc != ETIMEDOUT)
        break;
    if (sync_pending || gtns - tstart >= sleep_ns)
        break;
    if (wal_dirty())
        break;
}
```

Reply:
Fixed, reads much better. Thanks!